src/main/java/io/ebean/config/properties/YamlLoader.java

```diff
@@ -27,6 +28,12 @@ void load(InputStream is) {
       for (Object map : yaml.loadAll(is)) {
         loadMap((Map<String, Object>)map, null);
       }
+
+      try {
+        is.close();
+      } catch (IOException ioe) {
+        // Nothing, because that's what it was before this.
+      }
```

Comment:
It's not a good design pattern to close resource-parameters in methods. There are the following converntions (which may violated in certain cases)
- If you pass in a resource, the caller is responsible to close it
- If you wrap a stream (e.g. `new BufferedInputStream(is)`), closing the outer stream will close also the inner.

You should fix that resource leak at the caller side, there is only one in io.ebean.config.properties.Loader:
```java
void loadYaml(String resourcePath, Source source) {
  if (yamlLoader != null) {
    try (InputStream is = resource(resourcePath, source) {
      yamlLoader.load(is);
    } catch (Exception e) {
      log.warn("Failed to read yml from:" + resourcePath, e);
    }
  }
}
```

Reply:
I was wondering about that, caller side it is.